### PR TITLE
QPT: QE-210 Get Auth Token Service

### PR DESCRIPTION
**Pull request by avattipalli**
Jira: [QE-210](https://jira.bigcommerce.com/browse/QE-210)

## What? Why?
As part of investigating [the right auth_token](https://jira.bigcommerce.com/browse/ANA-1803) to be passed for experiment service, I was able to find that we need to get the auth token by passing in username, password and use that to make api calls to experiment service. This change is to add the missing api call (getAuthToken) in our services.

If introducing a new structure or paradigm into the code or if adding a new helper or utility, use this as an opportunity to describe it.

## Affected areas?
Call out any changes that will impact, modify or alter the tests that will use the existing logic/codebase/domains.

For each affected area, indicate how your code will change it and how big the risk or impact is.

Do any of the tests needs to be updated potentially because of this change.

## Additional Deploy Steps? (Optional)
None

## How was it tested?

- [ ] Cloud Dev end-to-end - Describe Steps
- [x] Additional test cases (Optional)
- [ ] Integration - Describe Steps (Optional)

## Testing / Proof
...

## How can this be undone?
1. Revert this PR

ping @bigcommerce/billing-iam-testing

### CODE REVIEW

**Comment by avattipalli** [comment]
Once this change is merged, I will follow up with an update to the current implementation of experiment-service calls

File: src/global-constants.js [hunk 1]
```diff
@@ -6,6 +6,7 @@ module.exports = {
   DEFAULT_TIMEOUT: 10,
   EXPERIMENT_ENDPOINT : '/api/v2/Private/Experiment',
   PROVISIONING_REQUEST_ENDPOINT: 'api/v1/provisioning_requests/',
+  AUTH_TOKEN_ENDPOINT: '/api/v2/token',
```
Comment (aaron-humerickhouse) [inline src/global-constants.js:9]: AUTH_TOKEN is pretty vague, can it be a little more specific?  We have so many versions of authentication throughout the platform.

**Review by aaron-humerickhouse — COMMENTED** [review]
One question with a likely change, rest looks good

File: src/global-constants.js (hunk 1 above)
Comment (avattipalli) [reply src/global-constants.js:9]: That's a fair call. I struggled myself by assuming oh, this is the auth_token that has to be used and then it not being the right one. I will come up with a better name for it.

Comment (avattipalli) [reply src/global-constants.js]: This is tied to STORE_CREATION so I renamed it to `STORE_CREATE_AUTH_TOKEN_ENDPOINT`

File: src/global-constants.js [hunk 2]
```diff
@@ -6,6 +6,7 @@ module.exports = {
   DEFAULT_TIMEOUT: 10,
   EXPERIMENT_ENDPOINT : '/api/v2/Private/Experiment',
   PROVISIONING_REQUEST_ENDPOINT: 'api/v1/provisioning_requests/',
+  STORE_CREATE_AUTH_TOKEN_ENDPOINT: '/api/v2/token',
```
Comment (aaron-humerickhouse) [inline src/global-constants.js:9]: Maybe STORE_V2_AUTH_TOKEN_ENDPOINT?

https://github.com/bigcommerce/bigcommerce/blob/master/lib/Store/Api/Version/2/Resource/Token.php#L8

This token can be used fore more than just creating a store

Comment (avattipalli) [reply src/global-constants.js:9]: Yeah, that makes sense. Let me update that quick

**Review by aaron-humerickhouse — APPROVED** [review]

**Pull request merged**